pkg/distributor/distributor.go

```diff
@@ -322,6 +350,10 @@ func (d *Distributor) validateSeries(ts ingester_client.PreallocTimeseries, user
 		nil
 }
 
+func (d *Distributor) validateMetadata(m *ingester_client.MetricMetadata, userID string) error {
+	return validation.ValidateMetadata(d.limits, userID, m)
+}
+
 // Push implements client.IngesterServer
 func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*client.WriteResponse, error) {
```

Comment:
I plan to add another change that merges the two. Since ingesters receive a `WriteRequest`, they can just go together. I just wanted to avoid a big refactor upfront.

Reply:
Hmm, ok. `Push` is quite big at the moment and could use some trimming.